src/app/static/src/tests/components/genericChart/genericChart.test.ts

```diff
@@ -753,16 +731,13 @@ describe("GenericChart component", () => {
         });
     });
 
-    it("does not fetch default dataset on data source value change if it already exists in state", (done) => {
+    it("does not fetch default dataset on data source value change if it already exists in state", async () => {
         const state = {datasets};
         const wrapper = getWrapper(state);
-        setTimeout(() => {
-            wrapper.findAllComponents(DataSource)[0].vm.$emit("update", "dataset2");
-            setTimeout(() => {
-                expect(mockAxios.history.get.length).toBe(0);
-                done();
-            });
-        });
+        await flushPromises();
+        wrapper.findAllComponents(DataSource)[0].vm.$emit("update", "dataset2");
+        await nextTick();
```

Comment:
Should this be a flushPromises? It is in the above test, where you're testing for the positive case where an API fetch *is* done.

Reply:
yes nice spot! as time went on i started leaning more towards `flushPromises` so hopefully should not be a frequent problem! will change it

Reply:
done